packages/cli/src/modules/chat-hub/chat-hub.service.ts

```diff
@@ -922,23 +914,27 @@ export class ChatHubService {
 	}
 
 	/**
-	 * Build the active message chain ending to the message with ID `lastMessageId`
+	 * Build the message history chain ending to the message with ID `lastMessageId`
 	 */
-	private buildActiveMessageChain(messages: ChatHubMessage[], lastMessageId: ChatMessageId) {
-		const nodes = new Map(messages.map((m) => [m.id, m]));
+	private buildMessageHistory(
```

Comment:
unlike FE, which tries to solve the current discussion that includes a message this just solves the history of a given message - which we know/assume is the latest one in the context (from its usage)